api/v2/performanceprofile_validation.go

```diff
@@ -0,0 +1,278 @@
+limitations under the License.
+*/
+
+package v2
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	"github.com/openshift-kni/performance-addon-operators/pkg/controller/performanceprofile/components"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/klog"
+	kubeletconfigv1beta1 "k8s.io/kubelet/config/v1beta1"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+const (
+	hugepagesSize2M = "2M"
+	hugepagesSize1G = "1G"
+)
+
+var _ webhook.Validator = &PerformanceProfile{}
+
+// we need this variable only because our validate methods should have access to the client
+var validatorClient client.Client
+
+// InitValidatorClient defines the validator client that should be used under validator methods
+func InitValidatorClient(mgr ctrl.Manager) {
+	if validatorClient != nil {
+		return
+	}
+	validatorClient = mgr.GetClient()
+}
+
+// SetupWebhookWithManager enables Webhooks - needed for version conversion
+func (r *PerformanceProfile) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (r *PerformanceProfile) ValidateCreate() error {
+	klog.Infof("Create validation for the performance profile %q", r.Name)
+
+	return r.validateCreateOrUpdate()
+}
+
+// ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
+func (r *PerformanceProfile) ValidateUpdate(old runtime.Object) error {
+	klog.Infof("Update validation for the performance profile %q", r.Name)
+
+	return r.validateCreateOrUpdate()
+}
+
+func (r *PerformanceProfile) validateCreateOrUpdate() error {
+	var allErrs field.ErrorList
+
+	// validate node selector duplication
+	ppList := &PerformanceProfileList{}
+	if err := validatorClient.List(context.TODO(), ppList); err != nil {
+		allErrs = append(allErrs, field.InternalError(field.NewPath("spec.nodeSelector"), err))
+		return apierrors.NewInternalError(err)
+	}
+
+	allErrs = append(allErrs, r.validateNodeSelectorDuplication(ppList)...)
+
+	// validate basic fields
+	allErrs = append(allErrs, r.validateFields()...)
+
+	if len(allErrs) == 0 {
+		return nil
+	}
+
+	return apierrors.NewInvalid(
+		schema.GroupKind{Group: "performance.openshift.io", Kind: "PerformanceProfile"},
+		r.Name, allErrs)
+}
+
+// ValidateDelete implements webhook.Validator so a webhook will be registered for the type
+func (r *PerformanceProfile) ValidateDelete() error {
+	klog.Infof("Delete validation for the performance profile %q", r.Name)
+
+	// TODO(user): fill in your validation logic upon object deletion.
+	return nil
+}
+
+func (r *PerformanceProfile) validateNodeSelectorDuplication(ppList *PerformanceProfileList) field.ErrorList {
```

Comment:
next time please split the new code from the code movement (two separate commits)